cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -98,6 +98,17 @@ Result<std::shared_ptr<SchemaManifest>> GetSchemaManifest(
   return manifest;
 }
 
+bool isNan(const Scalar& value) {
+  if (value.type->Equals(*float32())) {
+    const FloatScalar& float_scalar = checked_cast<const FloatScalar&>(value);
+    return std::isnan(float_scalar.value);
```

Comment:
Do we know for sure that `float_scalar.valid` is true?

Reply:
Checking for validity now.